### DISCUSSION TITLE
Change Extension Gallery to Self-Hosted Not Works in Theia Blueprint

### DISCUSSION THREAD

**Discussion by zlweicoder** [eclipse-theia/theia#11012] [Q&A] [answered]
Hi,

I'm trying to change the extension gallery of [Theia Blueprint IDE](https://theia-ide.org/docs/blueprint_download) to my self-hosted open-vsx server. And I found the documentation [One can host own instance of a registry and configure VSX_REGISTRY_URL environment variable to use it.](https://github.com/eclipse-theia/theia/blob/master/packages/vsx-registry/README.md) , so I added the environment variable "VSX_REGISTRY_URL" in my local environment like :
![56JAOb0J0T](https://user-images.githubusercontent.com/13825714/162974289-a0aea0c4-9b78-4dcb-bcb8-a925d2c39385.png)

After doing that, I restarted Theia Blueprint IDE and found it still connect open-vsx.org.

Any help would be appreciated~

Thanks.

**Comment by vince-fugnitto** [comment, marked as answer]
@zlweicoder thank you for the discussion :+1: 

It seems to work fine for me when I tested (using `blueprint` as well), please note that in order to have access to the env variable you'll need to re-use the terminal where you set it:

I verified by:

- open a terminal
- perform `yarn` on the `theia-blueprint` repo
- in the same terminal setting a test env variable (ex: `export VSX_REGISTRY_URL="test"`)
- in the same terminal, start the application (ex: `(cd applications/electron/ && yarn start)`)
- when the application starts it used my `VSX_REGISTRY_URL`

My env in that terminal has:

```bash
$ printenv | grep "VSX"
VSX_REGISTRY_URL=test
```

**Reply by zlweicoder** [reply]
@vince-fugnitto  very appreciate for your kind help, I'm not using code to start the theia-blueprint ide, I just download the mac install dmg file as this link: https://theia-ide.org/docs/blueprint_download shown, is this the reason?

**Reply by msujew** [reply]
@zlweicoder The way Vince describes to set the env variable guarantees that it is available inside of the blueprint app. It might be that your environment doesn't correctly propagate the env variables to blueprint. Can you run `printenv` inside of the app using the built-in terminal and confirm that the `VSX_REGISTRY_URL` is correctly set?

**Reply by zlweicoder** [reply]
@msujew You are correct, the env variable is not shown in the built-in terminal of Theia-blueprint ide as blow:
![image](https://user-images.githubusercontent.com/13825714/163351809-031028c8-c3ee-4bfb-9d8e-1951c0e39572.png)

Could you please help to let me know how to change the Theia-blueprint app's env variable?

Many thanks!

**Reply by msujew** [reply]
Not too familiar with OSX (which I assume you are using based on the window frame?), but there seem to be some ways to globally set env variables, see [this](https://stackoverflow.com/a/3756686) or [this](https://stackoverflow.com/a/36161817) stackoverflow thread.

Generally, you can always also start the blueprint app from the terminal, where you've set the env variable. The terminal will then propagate the variables to the started app.

**Reply by zlweicoder** [reply]
Thanks a lot for you help!
